src/core/util/options.js

```diff
@@ -270,12 +270,21 @@ function normalizeProps (options: Object) {
  */
 function normalizeInject (options: Object) {
   const inject = options.inject
+  const normalized = {}
+  let val
   if (Array.isArray(inject)) {
-    const normalized = options.inject = {}
```

Comment:
you can keep this line (and move it outside of the if) and remove the last line `options.inject = normalized`